six/modules/c++/six.sicd/source/Grid.cpp

```diff
@@ -84,4 +471,831 @@ bool Grid::operator==(const Grid& rhs) const
         timeCOAPoly == rhs.timeCOAPoly &&
         row == rhs.row && col == rhs.col);
 }
+
+bool Grid::validateTimeCOAPoly(
+        const CollectionInformation& collectionInformation,
+        logging::Logger& log) const
+{
+    const RadarModeType& mode = collectionInformation.radarMode;
+
+    //2.1. Scalar TimeCOAPoly means SPOTLIGHT data
+    bool isScalar = true;
+    bool valid = true;
+
+    // I don't know that it's impossible for a one-degree polynomial to be expressed
+    // as a polynomial of higher order for whatever reason, so I'm checking each term
+    // manually
+    for (size_t ii = 0; ii <= timeCOAPoly.orderX(); ++ii)

```

Comment:
This seems like a good thing to make a member function of the `TwoD` class... maybe `isScalar()`?